crates/agent/src/api/public/status.rs

```diff
@@ -0,0 +1,80 @@
+use std::sync::Arc;
+
+use crate::api::error::ApiErrorExt;
+use crate::api::{ApiError, App, ControlClaims};
+use axum::extract::{Path, State};
+use axum::http::StatusCode;
+use axum::{Extension, Json};
+// axum_extra's `Query` is needed here because unlike the one from `axum`, it
+// handles multiple query parameters with the same name
+use axum_extra::extract::Query;
+use chrono::{DateTime, Utc};
+use itertools::Itertools;
+use models::status::{self, StatusResponse};
+use models::Id;
+
+/// Query parameters for the status endpoint
+#[derive(Debug, serde::Deserialize, schemars::JsonSchema)]
+pub struct StatusQuery {
+    /// The catalog name of the live spec to get the status of
+    pub name: Vec<String>,
```

Comment:
The way to request multiple statuses would be `?name=foo&name=bar`, which is why the name is singular. This kinda seems like the best way to do it, since otherwise we'd need to somehow encode multiple values in one parameter, and we'd _still_ need to deal with the possibility that the request might provide a query parameter multiple times. IMHO just providing the query parameter multiple times is the most straight forward way to do this, though I'd want to hear if you disagree.

Reply:
Yeah IMO `?name=foo&name=bar` is perfectly fine, and agreed `name` names sense in that context